src/Visualization/Python/PlotSlice.py

```diff
@@ -0,0 +1,220 @@
+        "data. Specify as comma-separated list, e.g. '0,1,0' so the y-axis "
+        "is the vertical axis of the plot."
+    ),
+)
+@click.option(
+    "--num-samples",
+    "-N",
+    type=int,
+    nargs=2,
+    default=(200, 200),
+    show_default=True,
+    help=(
+        "Number of uniformly spaced samples along each direction of the slice "
+        "to which volume data is interpolated."
+    ),
+)
+@click.option(
+    "--num-threads",
+    "-j",
+    type=int,
+    show_default="all available cores",
+    help=(
+        "Number of threads to use for interpolation. Only available if compiled"
+        " with OpenMP. Parallelization is over volume data files, so this only"
+        " has an effect if multiple files are specified."
+    ),
+)
+@apply_stylesheet_command()
+@show_or_save_plot_command()
+def plot_slice_command(
+    h5_files,
+    subfile_name,
+    obs_id,
+    obs_time,
+    var_name,
+    slice_origin,
+    slice_extent,
+    slice_normal,
+    slice_up,
+    num_samples,
+    num_threads,
+):
+    """Plot variables on a slice through volume data
+
+    Interpolates the volume data in the H5_FILES to a slice and plots the
+    selected variables. You choose the slice by specifying three corners of a
+    rectangle.
+    """
+    # Interpolate the selected variable to the slice
+    if (
+        slice_origin is None
+        or slice_extent is None
+        or slice_normal is None
+        or slice_up is None
+    ):
+        raise click.UsageError(
+            f"Specify all slice parameters: '--slice-[origin,extent,normal,up]'"
+        )
+    target_coords = points_on_slice(
+        slice_origin, slice_extent, slice_normal, slice_up, num_samples
+    )
+    dim = target_coords.shape[0]
+    assert dim == 3, "Only 3D slices are supported"
+    data = np.array(
+        interpolate_to_points(
+            h5_files,
+            subfile_name=subfile_name,
+            observation_id=obs_id,
+            tensor_components=[var_name],
+            target_points=target_coords.reshape(3, np.prod(num_samples)),
+            num_threads=num_threads,
+        )[0]
+    ).reshape(num_samples)
+
+    # Set axes for plot. For plotting along an axis, use the axis coordinate.
+    # Otherwise, use the affine parameter.
+    def coord_axis(vec):
+        nonzero_entries = np.nonzero(vec)[0]
+        if len(nonzero_entries) == 1:
+            return nonzero_entries[0]
+        return None
+
+    y_axis = coord_axis(slice_up)
+    x_axis = coord_axis(np.cross(slice_up, slice_normal))
+    if x_axis is not None:
+        x = target_coords[x_axis]
+        x_label = "xyz"[x_axis]
+    else:
+        x = np.linspace(0, 1, num_samples[0])
+        x_label = None
+    if y_axis is not None:
+        y = target_coords[y_axis]
+        y_label = "xyz"[y_axis]
+    else:
+        y = np.linspace(0, 1, num_samples[1])
+        y_label = None
+
+    # Plot the slice
+    plt.contourf(x, y, data)
+    contours = plt.contour(
```

Comment:
Probably not for this PR, but we may want to offer other types of plots besides contours.

Reply:
If someone wants to do anything more than look at a quick slice through the data, then they can do that themselves in Python. We can't build all the possible things into the CLI, it's just to get a quick idea of what's going on with a simulation.